src/doppler/sinks/syslogwriter/https_writer_test.go

```diff
@@ -83,7 +91,76 @@ var _ = Describe("HttpsWriter", func() {
 			parsedTime, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
 			_, err = w.Write(standardErrorPriority, []byte("Message"), "just a test", "TEST", parsedTime.UnixNano())
 			Expect(err).ToNot(HaveOccurred())
+		})
 
+		Describe("idle connection behavior", func() {
+			var (
+				receivedChan chan []byte
+				waitChan     chan struct{}
+				listener     *historyListener
+			)
+
+			BeforeEach(func() {
+				receivedChan = make(chan []byte, 1)
+				waitChan = make(chan struct{})
+
+				l, err := net.Listen("tcp", "127.0.0.1:0")
+				Expect(err).NotTo(HaveOccurred())
+
+				listener = &historyListener{Listener: l}
+				server.Listener = listener
+
+				slowHandler := func(_ http.ResponseWriter, r *http.Request) {
```

Comment:
No. Until this test, the `ServeHTTP` function was really simple and we only needed to inspect which requests were being sent; the single `requestChan` was an acceptable interface. 

With your change, there's a lot more coordination between the test and the handler, so this test needs to use a cleaner interface (one that abstracts away all the coordination and channels).